yt_dlp/YoutubeDL.py

```diff
@@ -2605,6 +2606,7 @@ def _calc_cookies(self, url):
 
     def _sort_thumbnails(self, thumbnails):
         thumbnails.sort(key=lambda t: (
+            t.get('id') == self.params.get('thumbnail_id') if t.get('id') is not None else False,
```

Comment:
Don't change sorting. Select the correct one in [_write_thumbnails](https://github.com/yt-dlp/yt-dlp/blob/3905f64920ed078d9eeb5640884f5854e01d744d/yt_dlp/YoutubeDL.py#L4373) instead